ocs_ci/utility/aws.py

```diff
@@ -427,6 +458,159 @@ def get_ec2_instance_volumes(self, instance_id):
+            instance_id (str): Required instance to get security groups from it
+
+        Returns:
+            list: all_sg_ids: all attached security groups id.
+
+        """
+        ec2_instance = self.get_ec2_instance(instance_id)
+        all_sg_ids = [sg.get('GroupId') for sg in ec2_instance.security_groups]
+
+        return all_sg_ids
+
+    def create_security_group(self, group_name, dict_permissions, vpc_id):
+        """
+        Create security group with predefined group name and permissions
+
+        Args:
+            group_name (str): Group name (aws tag: "Group Name")
+            dict_permissions (dict): The security group's inbound/outbound permissions
+            vpc_id(str): For group to be attached
+
+        Returns:
+            str: newly created security group id
+
+        """
+        instance_response = self.ec2_client.create_security_group(
+            GroupName=group_name,
+            Description='This group created by method:aws.create_security_group',
+            VpcId=vpc_id
+        )
+
+        security_group_id = instance_response['GroupId']
+        logger.info(f'Security Group Created {security_group_id} in vpc {vpc_id}')
+
+        data = self.ec2_client.authorize_security_group_ingress(
+            GroupId=security_group_id,
+            IpPermissions=[dict_permissions]
+        )
+        logger.info(f'Ingress Successfully Set {data}')
+        return security_group_id
+
+    def append_security_group(self, security_group_id, instance_id):
+        """
+        Append security group to selected ec2 nodes
+
+        Args:
+            instance_id(str): Instances to attach security group
+            security_group_id(str): Security group to attach
+
+            print out: security group <id> added to selected nodes
+
+        """
+        ec2_instance = self.get_ec2_instance(instance_id)
+        logger.info(f"ec2_instance id is: {ec2_instance.id}")
+        all_sg_ids = [sg.get('GroupId') for sg in ec2_instance.security_groups]
+        if security_group_id not in all_sg_ids:
+            all_sg_ids.append(security_group_id)
+            ec2_instance.modify_attribute(Groups=all_sg_ids)
+
+        logger.info(f"Security Group {security_group_id} added to selected node")
+
+    def remove_security_group(self, security_group_id, instance_id):
+        """
+        Remove security group from selected ec2 instance (by instance id)
+        print out: security group <id> removed from selected nodes
+
+        Args:
+            security_group_id (str): Security group to be removed
+            instance_id (str): Instance attached with selected security group
+
+        """
+        ec2_instance = self.get_ec2_instance(instance_id)
+        logger.info(f"ec2_instance id is: {ec2_instance.id}")
+        all_sg_ids = self.get_security_groups_by_instance_id(instance_id)
+        for sg in all_sg_ids:
+            if sg == security_group_id:
+                all_sg_ids.remove(security_group_id)
```

Comment:
If the remove fails, does it raise an exception? Otherwise, an assertion is needed